source/GameMode.h

```diff
@@ -36,8 +36,12 @@ class GameMode {
 	/** The Ship model */
 	std::shared_ptr<ShipModel> ship;
 
-	bool host = true;
+	/** Local record of Player ID */
 	int playerID;
+	/** Local record of Room ID */
+	std::string roomId;
+	/** Local record of Network Status */
+	MagicInternetBox::MatchmakingStatus status;
```

Comment:
Why does there need to be a local copy of the status?

Reply:
I pass it to the gamegraphroot through a setter instead of passing mib to the gamegraphroot

Reply:
Right, but why does `GameMode` need its own local copy? It's only ever used in that big switch statement in `update()`; why can't you just switch directly on `net->matchStatus()`?

Reply:
Hmm, I think I originally planned on using ifs, but then realized a switch block would be better and never changed it.